### ISSUE TITLE
target of 'resource-file' in config.xml path correct?

### ISSUE THREAD

**Issue by fquirin** [apache/cordova-cli#582]
# Bug Report (or feature?)

## Problem

When using `target` in `<resource-file>` tag of `config.xml` the path location is unclear looking at the documentation.

### What is expected to happen?

When using `<resource-file src="FooPluginStrings.xml" target="res/values/FooPluginStrings.xml" />` as [described in the docs](https://cordova.apache.org/docs/en/11.x/config_ref/index.html#resource-file) I'd expect the file to end up in the according resources folder of the platform, in this case `app/src/main/res/values/FooPluginStrings.xml`.

### What does actually happen?

The file ends up in the top-level project folder for Android.

## Information

I'm not sure if this is a bug or a feature but the documentation is at least misleading. `res/values/...xml` is the classic resource path for Android platform but you have to use `app/src/main/res/values/...xml` as `target` to actually get this path. In the example this should be:
```
<resource-file src="FooPluginStrings.xml" target="app/src/main/res/values/FooPluginStrings.xml" />
```

In addition it seems the behavior in `config.xml` and `plugin.xml` is different 🤔. Plugin resource files seems to be placed in the correct folder without adding `app/src/main`.

### Command or Code

Put a resource in your config.xml and build the Cordova project, as seen in the docs:
```
<resource-file src="FooPluginStrings.xml" target="res/values/FooPluginStrings.xml" />
```

### Environment, Platform, Device

Debian 9.5 terminal (Windows 11 WSL2), x86_64

### Version information

Cordova: 11.0.0, Android platform 11.0.0
Other Frameworks: none
Operating System: Debian 9.5 on Windows 11 WSL2

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

**Comment by BigBalli** [comment]
This is actually expected behavior, though the docs are definitely misleading. In `config.xml`, the `target` path for `<resource-file>` is relative to the platform project root (e.g. the root of the Android Gradle project), not relative to the Android app source directory. So you do need the full `app/src/main/res/values/FooPluginStrings.xml` path.

The reason `plugin.xml` behaves differently is that the plugin system has its own resource mapping logic that knows about the Android project structure and resolves `res/values/...` to the right place automatically.

So your workaround is correct:
```xml
<resource-file src="FooPluginStrings.xml" target="app/src/main/res/values/FooPluginStrings.xml" />
```

This might be worth a docs clarification PR if anyone wants to pick it up. Were you able to get things working with the full path?

**Comment by breautek** [comment]
I would agree that the document is incorrect. In more complex projects you may have multiple modules, not just the `app` module... so it would be unwise to assume the app module path.
